### DISCUSSION TITLE
Yaboot or Grub

### DISCUSSION THREAD

**Discussion by EchedelleLR** [AOSC-Dev/aosc-os-abbs#3414] [Q&A] [answered]
After the initialization of the RAM Disk in the powerpc guide, the installation of yaboot at the Bootstrap partition is requested. However, during the RAM Disk step, I checked how grub entries were added at the Grub menu and, as expected, it seems Yaboot is not installed in the provided XFCE tarball for ppc64, only Grub.

What should I do?.

**Comment by MingcongBai** [comment, marked as answer]
Jesus. Sorry for all the trouble you're running into - but at this point I think I am comfortable enough to recommend GRUB as the default (documentation work to follow if it works out here). It should be as simple as `grub-install --target=powerpc-ieee1275 --no-nvram` and `grub-mkconfig -o /boot/grub/grub.cfg`.

Also, I believe you will have to uncomment `#GRUB_TERMINAL_OUTPUT=console`.

Please let me know how this goes.

**Comment by MingcongBai** [comment]
I will also transfer this issue to AOSC-Dev/aosc-os-abbs.

**Comment by EchedelleLR** [comment]
Where should I install grub? In the "disk" itself as I do in other distributions or (as its Apple Partition Table type) in a specific partition?

Sorry, I am a bit new with this as only tried FreeBSD with a successful installation.

**Comment by MingcongBai** [comment]
You shouldn't need to specify a disk with this platform.

**Comment by EchedelleLR** [comment]
Then something is wrong in the line. Grub just points me to check the help and usage but I am sure the line is written correctly.

**Comment by MingcongBai** [comment]
Did it tell you anything about what parametre is wrong?

**Comment by EchedelleLR** [comment]
No, just to check that.

**Comment by MingcongBai** [comment]
Could you try appending `/dev/sda1`, assuming GRUB is to be installed on the main drive?

**Comment by EchedelleLR** [comment]
But isn't the first partition of a Apple table intouchable because it's where the partitions are defined? The second one, /dev/sda2 its dedicated to that I think as yaboot was requested to be installed there. But I am not sure.

root@lubuntu [ / ] ! grub-install --target=powerpc-ieee1275 --no-vram
grub-install: unrecognized option '--no-vram'
Try 'grub-install --help' or 'grub-install --usage' for more information.
root@lubuntu [ / ] ! grub-install --usage
Usage: grub-install [-vs?V] [-d DIR] [-k FILE] [--compress=no|xz|gz|lzo]
            [--directory=DIR] [--fonts=FONTS] [--install-modules=MODULES]
            [--pubkey=FILE] [--locale-directory=DIR] [--locales=LOCALES]
            [--modules=MODULES] [--themes=THEMES] [--verbose] [--allow-floppy]
            [--boot-directory=DIR] [--bootloader-id=ID]
            [--core-compress=xz|none|auto] [--disk-module=MODULE]
            [--efi-directory=DIR] [--force] [--force-file-id]
            [--label-bgcolor=COLOR] [--label-color=COLOR] [--label-font=FILE]
            [--macppc-directory=DIR] [--no-bootsector] [--no-nvram]
            [--no-rs-codes] [--product-version=STRING] [--recheck]
            [--removable] [--skip-fs-probe] [--target=TARGET] [--help]
            [--usage] [--version] [OPTION] [INSTALL_DEVICE]

This is what it says

**Comment by EchedelleLR** [comment]
Okay, doesnt matter, was my fault. I just copied your line and worked ;-;

**Comment by MingcongBai** [comment]
I was going to say that you mis-typed `--no-nvram`. No worries XD.

**Comment by EchedelleLR** [comment]
Oh yes, jajajajaja.

**Comment by EchedelleLR** [comment]
I need to ask other question, can I do it here or should I open other issue as expected?

**Comment by MingcongBai** [comment]
> I need to ask other question, can I do it here or should I open other issue as expected?

Why don't we open an issue simply titled "PPC64 Testing and Issues" or something along that line so we can assemble a little checklist to get them sorted?

Also if you'd like to help, please feel free to join us, either on GitHub or Telegram (if you have it), I can provide all necessary training for packaging and distribution maintenance.

**Comment by EchedelleLR** [comment]
Okay, this one is related: since this iMac is using a ATI Radeon video card, radeon.agpmode=-1 set in the bootloader should be preferable but I have no idea if I should add any other config being Grub.

**Comment by EchedelleLR** [comment]
Sorry, with this old firefox in the lubuntu live image I can't update the view automatically to see new posts easily. Yes, would be preferable but I think you could do it better to create the desired list. I could add using posts what I am checking at the moment. 

Away from that, about packaging and maintaining the distribution I would be glad but I would like to ensure a successful installation first.

**Comment by MingcongBai** [comment]
> Okay, this one is related: since this iMac is using a ATI Radeon video card, radeon.agpmode=-1 set in the bootloader should be preferable but I have no idea if I should add any other config being Grub.

Nope, that should be all you need.

**Comment by EchedelleLR** [comment]
So directly I add it to the grub conf. Okay.

**Comment by MingcongBai** [comment]
To `/etc/default/grub`, yes.

**Comment by MingcongBai** [comment]
> Sorry, with this old firefox in the lubuntu live image I can't update the view automatically to see new posts easily. Yes, would be preferable but I think you could do it better to create the desired list. I could add using posts what I am checking at the moment.

Please refer to https://github.com/AOSC-Dev/aosc-os-abbs/issues/2108 for other issues.

> Away from that, about packaging and maintaining the distribution I would be glad but I would like to ensure a successful installation first.

Sure thing, hopefully we can get this sorted out, and again, I am very sorry for all the trouble.

**Comment by EchedelleLR** [comment]
> Please refer to #2108 for other issues.

Okay and thank you for the help.

> To /etc/default/grub, yes.

Yes, I told that to confirm you were not telling me to not adding it.

**Comment by MingcongBai** [comment]
> Yes, I told that to confirm you were not telling me to not adding it.

Got it.

**Comment by EchedelleLR** [comment]
A good reference being found at https://wiki.adelielinux.org/wiki/Package:GRUB2

**Comment by MingcongBai** [comment]
We are now using GRUB for Retro distributions. Distributions and documentation updates to come.

**Comment by EchedelleLR** [comment]
:3
